src/Orchard.Web/Modules/Orchard.Localization/Services/LocalizationService.cs

```diff
@@ -29,11 +29,17 @@ public class LocalizationService : ILocalizationService {
             if (localized == null)
                 return null;
 
+            if (localized.Record.CultureId == cultureRecord.Id) {
```

Comment:
Here you can check for the record equality directly like `localized.Record == cultureRecord` but it's not very important, this change is already nice.